ingestion/src/metadata/ingestion/source/messaging/common_broker_source.py

```diff
@@ -199,37 +195,56 @@ def _parse_topic_metadata(self, topic_name: str) -> Optional[Schema]:
             )
         return None
 
-    def _get_sample_data(self, topic_name):
-        sample_data = []
-        try:
-            self.consumer_client.subscribe([topic_name.__root__])
-            logger.info(
-                f"Broker consumer polling for sample messages in topic {topic_name.__root__}"
-            )
-            messages = self.consumer_client.consume(num_messages=10, timeout=10)
-        except Exception as exc:
-            logger.debug(traceback.format_exc())
-            logger.warning(
-                f"Failed to fetch sample data from topic {topic_name.__root__}: {exc}"
-            )
-        else:
-            if messages:
-                for message in messages:
-                    try:
-                        sample_data.append(
-                            str(
-                                self.consumer_client._serializer.decode_message(  # pylint: disable=protected-access
-                                    message.value()
+    def yield_topic_sample_data(
+        self, topic_details: BrokerTopicDetails
+    ) -> TopicSampleData:
+        """
+        Method to Get Sample Data of Messaging Entity
+        """
+        if (
+            self.context.topic
+            and self.context.topic.messageSchema
+            and self.context.topic.messageSchema.schemaType.value
+            == SchemaType.Avro.value
```

Comment:
Why do we need it to be Avro? 🤔

Reply:
not sure why that's the case, I just translated [this](https://github.com/open-metadata/OpenMetadata/blob/831edcf0026e6247a660f03ddde9b91b1756b7ae/ingestion/src/metadata/ingestion/source/messaging/common_broker_source.py#L130) logic here 😅

Reply:
not sure if we have a way to test it out with JSON Schema and Protobuf, as we support those type of messages 🤔

Reply:
yup will check if I can test it with protobuf